ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/OpenSSL/tasks/main.yml

```diff
@@ -1,64 +1,184 @@
     - ansible_architecture == "x86_64"
-    - ((openssl_version.stdout == '') or ((openssl_version.stdout != '') and (openssl_version.stdout | version_compare('1.1.0', operator='lt', strict=True))))
+    - openssl_version.stdout == ''
+  tags: openssl
+
+# add to PATH
+
+- name: Set OpenSSL path command on CentOS or RHEL
+  set_fact:
+    path_commad: pathmunge /usr/local/openssl-{{ openssl_latest_release }}/bin
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
   tags: openssl
 
-- name: Remove downloaded packages for OpenSSL v1.1.0g
+- name: Set OpenSSL path command on SLES or Ubuntu
+  set_fact:
+    path_commad: export PATH=/usr/local/openssl-{{ openssl_latest_release }}/bin:$PATH
+  when:
+    - (ansible_distribution == "SLES" or  ansible_distribution == "Ubuntu")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  tags: openssl
+
+- debug: var=path_commad
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  tags: openssl
+
+- name: Remove /etc/profile.d/openssl*.sh
+  shell: find /etc/profile.d/ -type f -name openssl\*.sh | grep -v {{ openssl_latest_release }} | xargs rm -f
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  ignore_errors: True
+  tags: openssl
+
+- name: Create /etc/profile.d/openssl-{{ openssl_latest_release }}.sh
+  copy:
+    dest: /etc/profile.d/openssl-{{ openssl_latest_release }}.sh
+    content: "{{ path_commad }}"
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  tags: openssl
+
+# configure link libraries
+
+- name: Remove /etc/ld.so.conf.d/openssl*.conf
+  shell: find /etc/ld.so.conf.d/ -type f -name openssl\*.conf | grep -v {{ openssl_latest_release }} | xargs rm -f
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
+    - ansible_architecture == "x86_64"
+    - openssl_version.stdout == ''
+  ignore_errors: True
+  tags: openssl
+
+- name: Create /etc/ld.so.conf.d/openssl-{{ openssl_latest_release }}.conf
+  copy:
+    dest: /etc/ld.so.conf.d/openssl-{{ openssl_latest_release }}.conf
+    content: /usr/local/openssl-{{ openssl_latest_release }}/lib
+  when:
+    - (ansible_distribution == "RedHat" or ansible_distribution == "CentOS" or ansible_distribution == "Ubuntu" or ansible_distribution == "SLES")
```

Comment:
Can you specify which os / release variants you tested this against?

Reply:
centos 6, rhel6, ubuntu14, ubuntu16, sles12